src/rt/cast_.d

```diff
@@ -13,6 +13,15 @@
  */
 module rt.cast_;
 
+// because using == does a dynamic cast, but we
+// are trying to implement dynamic cast.
+bool compareClassInfo(ClassInfo a, ClassInfo b)
+{
+    if (a is b)
+        return true;
+    return (a && b) && a.info.name == b.info.name;
```

Comment:
When would the ClassInfo ever be null?

Reply:
I vaguely recall seeing it happen in December when working on this originally but I don't remember where. Possible I was just being defensive too.